cmd/nerdctl/login_unix.go

```diff
@@ -46,3 +48,26 @@ func readPassword() (string, error) {
 
 	return string(bytePassword), nil
 }
+
+func readUsername() (string, error) {
+	var fd *os.File
+	if term.IsTerminal(syscall.Stdin) {
+		fd = os.Stdin
+	} else {
+		tty, err := os.Open("/dev/tty")
```

Comment:
Shouldn’t we just read stdin even when it is not a terminal?

Reply:
I would like to know the use case that one would pass username **and** password by stdin and why, then password alone has such case, because of secret that is passed but username as stdin? docker login has only password stdin option.
cat secret-password | docker login --username foo --password-stdin

Reply:
adding username to stdin is not intuitive without `username-stdin` which is rarely used.

Reply:
So, `readUsername()` should return an error when stdin is not a terminal?
I don't think we have to open `/dev/tty` here.

Reply:
Added a follow-up commit to avoid /dev/tty